To: vim_dev@googlegroups.com
Subject: Patch 7.3.118
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.118
Problem:    Ruby uses SIGVTALARM which makes Vim exit. (Alec Tica)
Solution:   Ignore SIGVTALARM. (Dominique Pelle)
Files:	    src/os_unix.c


*** ../vim-7.3.117/src/os_unix.c	2010-12-17 16:27:10.000000000 +0100
--- src/os_unix.c	2011-02-09 18:19:57.000000000 +0100
***************
*** 283,289 ****
  #ifdef SIGTERM
      {SIGTERM,	    "TERM",	TRUE},
  #endif
! #ifdef SIGVTALRM
      {SIGVTALRM,	    "VTALRM",	TRUE},
  #endif
  #if defined(SIGPROF) && !defined(FEAT_MZSCHEME) && !defined(WE_ARE_PROFILING)
--- 283,289 ----
  #ifdef SIGTERM
      {SIGTERM,	    "TERM",	TRUE},
  #endif
! #if defined(SIGVTALRM) && !defined(FEAT_RUBY)
      {SIGVTALRM,	    "VTALRM",	TRUE},
  #endif
  #if defined(SIGPROF) && !defined(FEAT_MZSCHEME) && !defined(WE_ARE_PROFILING)
***************
*** 1107,1113 ****
   * On Linux, signal is not always handled immediately either.
   * See https://bugs.launchpad.net/bugs/291373
   *
!  * volatile because it is used in in signal handler sigcont_handler().
   */
  static volatile int sigcont_received;
  static RETSIGTYPE sigcont_handler __ARGS(SIGPROTOARG);
--- 1107,1113 ----
   * On Linux, signal is not always handled immediately either.
   * See https://bugs.launchpad.net/bugs/291373
   *
!  * volatile because it is used in signal handler sigcont_handler().
   */
  static volatile int sigcont_received;
  static RETSIGTYPE sigcont_handler __ARGS(SIGPROTOARG);
*** ../vim-7.3.117/src/version.c	2011-02-09 17:42:53.000000000 +0100
--- src/version.c	2011-02-09 18:46:53.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     118,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
221. Your wife melts your keyboard in the oven.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
